### PR TITLE
feat: report create index merge progress

### PR DESCRIPTION
**Pull request by hfutatzhanghb**
## Summary

Report distributed CREATE INDEX progress using the independent segment architecture now on main.

PR #701 replaced the old shared UUID plus mergeIndexMetadata flow with one uncommitted Lance segment per Spark task followed by an atomic commitExistingIndexSegments call. Lance Core does not expose a progress callback on createIndex or segment commit, so this implementation observes progress at the Spark driver boundary as successful segment partitions return.

## Changes

- Add Spark SQL metrics for eager distributed segment builds:
  - `index build completed segments`
  - `index build total segments`
- Execute segment tasks with the SparkContext runJob driver result handler so progress is visible before the full job result is available.
- Count each successful Spark partition once, preserving the independent segment retry and speculation isolation model.
- Keep progress metric and logging failures informational so they cannot fail index creation.
- Log segment build start and completion progress on the driver.
- Document the segment progress behavior.
- Add pure JVM tests for progress accounting, duplicate partition handling, failure isolation, and metric selection.
- Extend the existing real Lance zonemap segment integration test to verify completed and total metric values.

## Scope

Progress is reported for eager segment builds: zonemap, bitmap, label_list, ngram, bloomfilter, rtree, FTS or inverted, and BTree fragment mode. Range-mode BTree keeps its dedicated preprocessed-data path. Deferred and empty-table builds do not launch segment tasks.

This does not restore the removed mergeIndexMetadata path and does not change the atomic segment commit behavior.

## Validation

- `./mvnw spotless:check`
- `./mvnw compile -pl lance-spark-3.5_2.12 -am`
- `./mvnw compile -pl lance-spark-3.5_2.13 -am`
- `./mvnw test -pl lance-spark-3.5_2.12 -Dtest=IndexUtilsTest` — 29 tests passed
- `./mvnw test -pl lance-spark-3.5_2.13 -Dtest=IndexUtilsTest` — 29 tests passed

The local native segment integration run could not start because the installed Lance artifact does not include `darwin-x86-64/liblance_jni.dylib`; the GitHub Actions Spark matrix is the native validation gate.

### CODE REVIEW

**Comment by hfutatzhanghb** [comment]
Hi @fangbo @LuciferYang . Could you please help review this simple PR when have free time? Thanks very much!

File: lance-spark-base_2.12/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AddIndexExec.scala [hunk 1]
```diff
@@ -290,6 +305,13 @@ case class AddIndexExec(
       UTF8String.fromString(indexName))))
   }
 
+  private def createIndexMergeProgress(): IndexBuildProgress =
+    new SparkIndexBuildProgress(
+      indexName,
+      completedDelta => metrics("indexMergeCompletedUnits").add(completedDelta),
```
Comment (fangbo) [inline lance-spark-base_2.12/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AddIndexExec.scala]: `indexMergeCompletedUnits` is duplicated with 

```java
override lazy val metrics: Map[String, SQLMetric] = Map(
    "indexMergeCompletedUnits" ->
      SQLMetrics.createMetric(sparkContext, "index merge completed work units"),
    "indexMergeTotalUnits" ->
      SQLMetrics.createMetric(sparkContext, "index merge total work units"))
```

This can be used as a variable ?

File: lance-spark-base_2.12/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AddIndexExec.scala [hunk 2]
```diff
@@ -290,6 +305,13 @@ case class AddIndexExec(
...
+      totalDelta => metrics("indexMergeTotalUnits").add(totalDelta),
```
Comment (fangbo) [inline lance-spark-base_2.12/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AddIndexExec.scala]: `indexMergeTotalUnits` same as `indexMergeCompletedUnits`

**Review by fangbo — COMMENTED** [review]
left some comments

**Comment by hfutatzhanghb** [comment]
@fangbo Thanks very much for review. Have fixed.

**Review by fangbo — APPROVED** [review]

**Comment by hfutatzhanghb** [comment]
conflicts fixed.

**Comment by hfutatzhanghb** [comment]
@fangbo Hi. Could you please take a look again when have free time? Thanks very much！

**Comment by fangbo** [comment]
@hfutatzhanghb I think the Lance Gatekeeper should run and approve.

**Comment by hfutatzhanghb** [comment]
What happens on Lance GateKeeper here? Hah.
image

**Comment by hfutatzhanghb** [comment]
Will open a new PR since Lance GateKeeper does not work well here.
